xray/segment.go

```diff
@@ -409,6 +428,16 @@ func (seg *Segment) root() *Segment {
 	return seg.parent.root()
 }
 
+// check the segment is dummy segment or not
+func (seg *Segment) isDummy() *Segment {
```

Comment:
This method seems not necessary. It is only called at 2 places when begin a segment or subsegment. Customer doesn't need to call it as well. I'd suggest replacing the caller code with
 ```
if !seg.ParentSegment.Sampled 
   seg.Dummy = true
```
BTW The method name is confusing. `isXYZ()` should returns a boolean. Now the return type is never used (see the github warn below)

Reply:
good suggestion!